### ISSUE TITLE
Ensure that there are no aliases with the same name as an ingredient (in a test)

### ISSUE THREAD

**Issue by pachun** [pachun/mix-me#12]
If one drink recipe includes the ingredient "Coke":

```typescript
const drinkWithCoke: DrinkRecipe = {
 // ...
  portionlessIngredients: [
    {
      type: DrinkRecipeIngredientType.Portionless,
      ingredient: "Coke",
      alternates: [],
    },
  ],
}
```

And another recipe includes the ingredient "Dr Pepper":

```typescript
const drinkWithDrPepper: DrinkRecipe = {
 // ...
  portionlessIngredients: [
    {
      type: DrinkRecipeIngredientType.Portionless,
      ingredient: "Dr Pepper",
      alternates: [],
    },
  ],
}
```

Then the state of the `ingredientToIngredientAliases.ts` file looks like this:

```typescript
const ingredientToIngredientAliases: IngredientToIngredientAliases = {
  "Coke": ["Coca-Cola", "Cocacola"],
  "Dr Pepper": ["Doctor Pepper"],
}
```

And the search keywords that will show the first drink recipe with coke (case insensitively) are:

```
coke, coca-cola, cocacola
```

And the keywords that will show the second drink recipe with Dr Pepper (case insensitively) are:

```
dr pepper, doctor pepper
```

If we decide that in the context of the second recipe, it'd be acceptable to use coke instead of Dr Pepper, we should not do this:

```
const ingredientToIngredientAliases: IngredientToIngredientAliases = {
  "Coke": ["Coca-Cola", "Cocacola", "Dr Pepper"],
  "Dr Pepper": ["Doctor Pepper"],
}
```

We should not do this for several reasons:

* Now every recipe that requires Coke will be displayed when "Dr Pepper" is searched. That's not right. (and even if it is, there's a better way to do this)
* Concerning the recipe we intended to update, searching "Doctor Pepper" will not reveal it in the list of recipes. That's not right either.

We should do it this way:

```typescript
const drinkWithDrPepper: DrinkRecipe = {
 // ...
  portionlessIngredients: [
    {
      type: DrinkRecipeIngredientType.Portionless,
      ingredient: "Dr Pepper",
      alternates: ["Coke"], // list it in the alternates
    },
  ],
}

const ingredientToIngredientAliases: IngredientToIngredientAliases = {
  "Coke": ["Coca-Cola", "Cocacola"],
  "Dr Pepper": ["Doctor Pepper"],
}
```

We _should_ do this because:

* Now only this recipe allows substituting Dr Pepper for Coke. (and if every recipe can subsitute Dr Pepper for coke, always include it in the alternates list - being explicit is better).
* Concerning the recipe we intended to update, searching any of `coke, coca-cola, cocacola, dr pepper, doctor pepper` will reveal our updated recipe in the search results.

TLDR:

* `ingredientAliases` are strings which mean the ingredient they're describing, written in a different way (eg `coke, coca-cola, cocacola` and `Dr Pepper, doctor pepper`
* `ingredient.alternates` are for listing ingredient substitutions (and then you get the benefit of all the alternate drinks aliases bringing up the recipe as well).

**Comment by pachun** [comment]
Closing in favor of https://github.com/pachun/mix-me/issues/15
